src/containers/qtreetbl.c

```diff
@@ -1002,7 +1002,7 @@ int qtreetbl_check(qtreetbl_t *tbl) {
     if (node_check_rule5(tbl, tbl->root, &path_len))
         return 1;
     
-    
+    printf("No rules are violated.\n"); 
```

Comment:
Was this for debugging purpose?

Reply:
Yeah, though this piece of code  with 3 empty lines looked like it was missing a print haha

Reply:
Oh, I see. This is probably fine. I see this prints out in the check and unit tests.

Reply:
maybe it'd be cleaner if we remove all the print() including L996 in the logic and handle it in the unit tests code if some debug output is necessary.

Reply:
I agree with you